spring-cloud-dataflow-admin/src/main/java/org/springframework/cloud/dataflow/admin/controller/StreamController.java

```diff
@@ -304,27 +354,22 @@ else if (isFirst) {
 		return moduleDeploymentProperties;
 	}
 
-	private boolean upstreamModuleHasPartitionInfo(StreamDefinition stream, ModuleDefinition currentModule,
-			Map<String, String> cumulatedDeploymentProperties) {
-		Iterator<ModuleDefinition> iterator = stream.getDeploymentOrderIterator();
-		while (iterator.hasNext()) {
-			ModuleDefinition module = iterator.next();
-			if (module.equals(currentModule) && iterator.hasNext()) {
-				ModuleDefinition prevModule = iterator.next();
-				Map<String, String> moduleDeploymentProperties = getModuleDeploymentProperties(prevModule, cumulatedDeploymentProperties);
-				return moduleDeploymentProperties.containsKey(BindingProperties.PARTITION_KEY_EXPRESSION) ||
-						moduleDeploymentProperties.containsKey(BindingProperties.PARTITION_KEY_EXTRACTOR_CLASS);
-			}
+	private boolean moduleHasPartitionInfo(ModuleDefinition module, Map<String, String> cumulatedDeploymentProperties) {
```

Comment:
for consistency with `isPartitionedConsumer(..)`, can we call this `isPartitionedProducer(..)`?